packages/aws-cdk-lib/custom-resources/lib/provider-framework/waiter-state-machine.ts

```diff
@@ -75,23 +123,61 @@ export class WaiterStateMachine extends Construct {
       },
     });
 
-    const resource = new CfnResource(this, 'Resource', {
-      type: 'AWS::StepFunctions::StateMachine',
-      properties: {
-        DefinitionString: definition,
-        RoleArn: role.roleArn,
-      },
+    const resource = new CfnStateMachine(this, 'Resource', {
```

Comment:
We previously used `CfnResource` here because we did not want to depend on `aws-stepfunctions` module (back in CDK v1 where individual modules were released. Now that they are both part of `aws-cdk-lib`, this change is ok. But there are comments on line 80 that need to be updated to reflect this. I would document that we previously used `CfnResource` to avoid depending on `aws-stepfunctions`, but now it is okay.